src/node/interfaces.cpp

```diff
@@ -88,30 +88,75 @@ class EVOImpl : public EVO
     {
         const CBlockIndex *tip = WITH_LOCK(::cs_main, return ::ChainActive().Tip());
         CDeterministicMNList mnList{};
-        if  (tip != nullptr && deterministicMNManager != nullptr) {
-            mnList = deterministicMNManager->GetListForBlock(tip);
+        if (tip != nullptr && m_context->dmnman != nullptr) {
+            mnList = m_context->dmnman->GetListForBlock(tip);
         }
         return {std::move(mnList), tip};
     }
+    void setContext(NodeContext* context) override
+    {
+        m_context = context;
+    }
+
+private:
+    NodeContext* m_context;
```

Comment:
```suggestion
    NodeContext* m_context{nullptr};
```
same for all similar cases

Reply:
Resolved in latest push